rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,327 @@
+    lp_token_addresses: set[str]
+    underlying_token: ChecksumEvmAddress
+
+
+def get_existing_pools(
+        cursor: 'DBCursor',
+        cache_type: Literal[CacheType.GEARBOX_POOL_ADDRESS],
+) -> set[ChecksumEvmAddress]:
+    """Returns all the gearbox pool rewards address stored in cache"""
+    return {
+        string_to_evm_address(address) for address in globaldb_get_general_cache_values(
+            cursor=cursor,
+            key_parts=(cache_type,),
+        )
+    }
+
+
+def save_gearbox_data_to_cache(
+        write_cursor: 'DBCursor',
+        database: Optional['DBHandler'],  # pylint: disable=unused-argument
+        new_data: list[GearboxPoolData],
+) -> None:
+    """Stores data about gearbox pools and their names in the global db cache tables."""
+    for pool in new_data:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.GEARBOX_POOL_ADDRESS,),
+            values=[pool.pool_address],
+        )
+        globaldb_set_unique_cache_value(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.GEARBOX_POOL_NAME, pool.pool_address),
+            value=pool.pool_name,
+        )
+        globaldb_set_unique_cache_value(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.GEARBOX_POOL_UNDERLYING, pool.pool_address),
+            value=pool.underlying_token,
+        )
+        if pool.farming_pool_addresses is not None:
+            for idx, farming_token in enumerate(pool.farming_pool_addresses):
+                globaldb_set_general_cache_values(
+                    write_cursor=write_cursor,
+                    key_parts=(CacheType.GEARBOX_POOL_FARMING_TOKENS, pool.pool_address, str(idx)),
+                    values=[farming_token],
+                )
+        if pool.lp_token_addresses is not None:
+            for idx, lp_token in enumerate(pool.lp_token_addresses):
+                globaldb_set_general_cache_values(
+                    write_cursor=write_cursor,
+                    key_parts=(CacheType.GEARBOX_POOL_LP_TOKENS, pool.pool_address, str(idx)),
+                    values=[lp_token],
+                )
+
+
+def read_gearbox_data_from_cache() -> tuple[dict[ChecksumEvmAddress, Any]]:
+    """Reads convex pools and names from global db cache tables."""
+    pools: dict[ChecksumEvmAddress, Any] = {}
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        for pool_address in get_existing_pools(
+            cursor=cursor,
+            cache_type=CacheType.GEARBOX_POOL_ADDRESS,
+        ):
+            if (pool_name := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.GEARBOX_POOL_NAME, pool_address),
+            )) is None:
+                continue
+            if (underlying_token := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.GEARBOX_POOL_UNDERLYING, pool_address),
```

Comment:
do you need to cache the underlying tokens? We already have the underlying table and you save it along the tokens

Reply:
removed not needed